common/persistence/clusterMetadataStore.go

```diff
@@ -131,12 +154,26 @@ func (m *clusterMetadataManagerImpl) SaveClusterMetadata(request *SaveClusterMet
 		return false, nil
 	}
 
-	return m.persistence.SaveClusterMetadata(&InternalSaveClusterMetadataRequest{ClusterMetadata: mcm, Version: request.Version})
+	return m.persistence.SaveClusterMetadata(&InternalSaveClusterMetadataRequest{
+		ClusterName:     request.ClusterName,
+		ClusterMetadata: mcm,
+		Version:         request.Version,
+	})
+}
+
+func (m *clusterMetadataManagerImpl) DeleteClusterMetadata(request *DeleteClusterMetadataRequest) error {
+	if request.ClusterName == m.currentClusterName {
+		return serviceerror.NewInvalidArgument("Cannot delete cluster metadata")
```

Comment:
message should be cannot delete current cluster metadata

Reply:
updated the error message